metadata_request.go

```diff
@@ -42,7 +42,7 @@ func (r *MetadataRequest) decode(pd packetDecoder, version int16) error {
 	} else {
 		topicCount := size
 		if topicCount == 0 {
-			return nil
+			goto SKIP
```

Comment:
why not return here?

Reply:
If the protocol version is higher than 3 (0.11.0.0), which is the case of version 1.0.0.0 (5) that supports the offline replicas request, the if block on the bottom of the function:
```golang
if r.Version > 3 {
   autoCreation, err := pd.getBool()
   if err != nil {
      return err
   }
   r.AllowAutoTopicCreation = autoCreation
}
```
But obviously I admit this is a potential point of improvement, I can try to refactor this to a more idiomatic way.

Reply:
yeah, please have a look, if it's can't be refactored, I am not super against GOTO's

Reply:
done